src/runtime/schema/schema.ts

```diff
@@ -110,11 +114,20 @@ export function create(appState: AppState): SchemaInternal {
       settings: settings,
       reset() {
         statefulNexusSchema.state.types = []
+        statefulNexusSchema.state.scalars = {}
         appState.schemaComponent.contextContributors = []
         appState.schemaComponent.plugins = []
+        appState.schemaComponent.scalars = {}
       },
-      assemble: (plugins) => {
+      beforeAssembly() {
+        appState.schemaComponent.scalars = {
+          ...Scalars.builtinScalars,
```

Comment:
Re my previous comment, I guess instead of this it would be calls to the public api in singleton constructor.

I've mentioned before though that the lack of separation between user calls to public api and internal calls to public api might bite us in the future... At the same time, we have the same/similar problem with plugins no? So maybe we should "share the problem" and solve (if ever needed) for both.